src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,88 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+    private static final int INDEX_OF_SUPPLY_AMOUNT = 0;
+    private static final int INDEX_OF_BUY_AMOUNT = 1;
+    private static final int INDEX_OF_OPERATION_TYPE = 0;
+    private static final int INDEX_OF_AMOUNT = 1;
+    private static final String COMMA_SEPARATOR = ",";
+    private static final String SUPPLY_TYPE_STRING = "supply";
+    private static final String BUY_TYPE_STRING = "buy";
+    private static final String RESULT_REPORT_STRING = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] data = readDataFromFile(fromFileName);
+        String report = createReportFromData(data);
+        writeReportIntoFile(report, toFileName);
```

Comment:
Good, but let's reorder the method's declaration in the code according to the order of their calls. It should be easy to read the code like a book from up to down.

Reorder so:
- readDataFromFile
- parseDataLine
- createReportFromData
- writeReportIntoFile
- deleteIfExists

Then take a look and think if it improved the readability. See Clean Code of Robert Martin for more details.